docs/python_support/reference.md

```diff
@@ -0,0 +1,769 @@
+
+ Check out the [quickstart guide](quickstart.md) for a simple example that shows how to use this feature.
+
+ ## Building Docker Image
+
+The publicly available `openvino/model_server` image on Docker Hub supports Python, but does not come with external modules installed. If Python is all you need then you can use the public image without modification. Otherwise, you will need to extend the public image with additional layers that install any modules required for your Python code to run. For example, let's say your code requires numpy.
+In that case, your Dockerfile may look like this:
+
+```dockerfile
+FROM openvino/model_server:latest
+USER root
+ENV LD_LIBRARY_PATH=/ovms/lib
+ENV PYTHONPATH=/ovms/lib/python
+RUN apt update && apt install -y python3-pip git
+RUN pip3 install numpy
+ENTRYPOINT [ `/ovms/bin/ovms` ]
+```
+
+You can also modify `requirements.txt` from our [python demos](../../demos/python_demos) and from repository top level directory run `make python_image` 
+
+## `OvmsPythonModel` class
+
+When deploying a Python node, the Model Server expects a Python file with an `OvmsPythonModel` class implemented:
+
+```python
+class OvmsPythonModel:
+
+    def initialize(self, kwargs):
+        """
+        `initialize` is called when model server loads graph definition. 
+        It allows to initialize and maintain state between subsequent execute() calls and even graph instances. 
+        For gRPC unary, graphs are recreated per request. 
+        For gRPC streaming, there can be multiple graph instances existing at the same time. 
+        OvmsPythonModel object is initialized with this method and then shared between all graph instances. 
+        Implementing this function is optional.
+
+        Parameters:
+        -----------
+        kwargs : dict
+            Available arguments:
+            * node_name: string
+                Name of the node in the graph
+            * input_streams: list of strings
+                List of input stream names defined for the node in graph
+                configuration
+            * output_streams: list of strings
+                List of output stream names defined for the node in graph
+                configuration
+        -----------
+        """
+        print("Running initialize...")
+
+    def execute(self, inputs):
+        """
+        `execute` is called in `Process` method of PythonExecutorCalculator
+        which in turn is called by the MediaPipe framework. How MediaPipe
+        calls the `Process` method for the node depends on the configuration
+        and the two configurations supported by PythonExecutorCalculator are:
+        
+        * Regular: `execute` runs every time the node receives all inputs with
```

Comment:
Unfortunately it might not be always true, example:

One python node with multiple inputs, but includes ImmediateInputStreamHandler

Isnt user allowed to send N<M inputs (where M is the number of all defined inputs) and the execute will still be triggered?